### PR TITLE
MET-64: Add validation to prevent removing units when measurement family is locked

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Add update validation on update of a locked measurement family.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added legacy Behats               | -
| Added acceptance tests            | Ok
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/config/services.yml [hunk 1]
```diff
@@ -63,3 +63,12 @@ services:
             - '%akeneo_measure.validation.measurement_family.families_max%'
         tags:
             - { name: 'validator.constraint_validator', alias: 'akeneo_measure.validation.measurement_family.count' }
+
+    # Validation
+    akeneo_measure.validation.measure_family.when_used_in_a_product_attribute_should_be_able_to_update_only_labels_and_symbol_and_add_units:
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/config/services.yml]:
```suggestion
    akeneo_measure.validation.measurement_family.when_used_in_a_product_attribute_should_be_able_to_update_only_labels_and_symbol_and_add_units:
```

Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Resources/config/services.yml]: done

File: src/Akeneo/Tool/Bundle/MeasureBundle/Validation/MeasurementFamily/WhenUsedInAProductAttributeShouldBeAbleToUpdateOnlyLabelsAndSymbolAndAddUnitsValidator.php [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    {
+        $isMeasureFamilyLockedForUpdates = $this->isThereAtLeastOneAttributeConfiguredWithMeasurementFamily
+            ->execute($saveMeasurementFamily->code);
+
+        if (!$isMeasureFamilyLockedForUpdates) {
+            return;
+        }
+
+        try {
+            $measurementFamily = $this->measurementFamilyRepository->getByCode(MeasurementFamilyCode::fromString($saveMeasurementFamily->code));
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Validation/MeasurementFamily/WhenUsedInAProductAttributeShouldBeAbleToUpdateOnlyLabelsAndSymbolAndAddUnitsValidator.php]:
```suggestion
            $dbMeasurementFamily = $this->measurementFamilyRepository->getByCode(MeasurementFamilyCode::fromString($saveMeasurementFamily->code));
```

It's just a suggestion. Idea is just to clearly say that it comes from storage, you can pick the naming you want :)

Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Validation/MeasurementFamily/WhenUsedInAProductAttributeShouldBeAbleToUpdateOnlyLabelsAndSymbolAndAddUnitsValidator.php]: well, in my opinion it does not really matter if it comes from storage or not. Once you know what is the Pattern repository, the naming is clear enough without it but I see what you meant.

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/translations/validators.en_US.yml [hunk 3]
```diff
@@ -5,12 +5,19 @@ pim_measurements:
                 pattern: 'This field can only contain letters, numbers, and underscores.'
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/translations/validators.en_US.yml]: Can you review this file plz ?

**Pull request merged**